### DISCUSSION TITLE
Additional SQL hits on every request

### DISCUSSION THREAD

**Discussion by samloft** [livewire/livewire#4068] [1. Help] [answered]
Hello, I'm hoping I have not misunderstood the mount method, but I was under the assumption from the docs that mount is called on initial page load and never again (Confirmed by using ray to say "Mounted" when it is called).

For example, having something like this in mount

`$this->foo = Foo::firstOrFail($id);`
`$this->foo->load(['bar' => function ($query) {
        $query->where('column', $this->foo->column);
    }]);
`

Then a button to open a modal with

`wire:click="$set('open', true)"`

I would expect to see things like session database hits, but I get eloquent queries being hit too, and worse than that, the SQL in the mount for the load, no longer does the closure

So instead of 

`select * from bar where column = 'foo'`

The SQL is 

`select * from bar`

Nothing in render except for loading the view and $this->foo is only populated in the mount method.

**Comment by joshhanley** [comment, marked as answer]
@samloft you are correct, mount does only run on the very first request and doesn't run again.

What you are seeing is Livewire's deserialisation process.

Livewire uses Laravel's QueueableCollection/QueueableEntity for serialising and deserialising Eloquent Collections/ Eloquent Models.

So what happens is Livewire only keeps track of the model class, ID, and any relations set on a model that has been assigned to a property (as that is what is defined by Laravel).

The way it restores the model is by retrieving it from the database, which is why you are seeing the database calls. It doesn't use the session for storing data, as it sends all state to the front end.

As such though, when it is retrieving the model from the database, it only know what relations were loaded on the model previously by name (so 'bar' in your example), as so calls `->load(['bar'])`. 

It has no way of knowing that you have a custom query inside your load method, hence why you aren't seeing the query be filtered.

Hope this helps!
